M2/Macaulay2/m2/document.m2

```diff
@@ -429,37 +429,46 @@ locate DocumentTag := tag -> new FilePosition from (
 -- helpers for the document function
 -----------------------------------------------------------------------------
 
+-- e.g. gb vs (codim, Ideal)
+getOptionBase := key -> (
+    if instance(key, Symbol) then key = value key;
+    try options key =!= null then key else try options key#0 =!= null then key#0 else key)
```

Comment:
Yes, unfortunately `options` isn't defined for a few things that appear in the documentation. I think mainly symbols and arrays? I could turn this into a method to avoid the tries, but was too lazy to figure out all the different cases ¯\_(ツ)_/¯

Reply:
Okay, gotcha.  But what about the `=?= null` stuff?  Regardless of whether that's true or false, we'll end up in the `then` clause:

```m2
i1 : try true then 1 else 2

o1 = 1

i2 : try false then 1 else 2

o2 = 1

i3 : try 1/0 then 1 else 2

o3 = 2
```

So should this just be `try options key then key else try options key#0 then key#0 else key`?

Reply:
Ah yes I messed it up. I'll fix it in a minute.